syntax = "proto3";

package order_management;

service OrderManagement {
    rpc CreateOrder(Order) returns (OrderResponse);
    rpc GetOrderStatus(OrderStatusRequest) returns (OrderStatusResponse);
}

message Order {
    string product_name = 1;
    string price = 2;
    string payment_gateway = 3;
    string card_brand = 4;
    string bank = 5;
    string shipping_direction = 6;
    string email = 7;
}

message OrderResponse {
    string order_id = 1;
}

message OrderStatusRequest {
    string order_id = 1;
}

message OrderStatusResponse {
    string status = 1;
}